beacon-chain/core/blocks/withdrawals.go

```diff
@@ -169,3 +139,41 @@ func ProcessWithdrawals(st state.BeaconState, withdrawals []*enginev1.Withdrawal
 	}
 	return st, nil
 }
+
+func BLSChangesSignatureBatch(
+	ctx context.Context,
+	st state.ReadOnlyBeaconState,
+	changes []*ethpb.SignedBLSToExecutionChange,
+) (*bls.SignatureBatch, error) {
+	// Return early if no changes
+	if len(changes) == 0 {
+		return bls.NewSet(), nil
+	}
+	batch := &bls.SignatureBatch{
+		Signatures: make([][]byte, len(changes)),
+		PublicKeys: make([]bls.PublicKey, len(changes)),
+		Messages:   make([][32]byte, len(changes)),
+	}
+	epoch := slots.ToEpoch(st.Slot())
+	domain, err := signing.Domain(st.Fork(), epoch, params.BeaconConfig().DomainBLSToExecutionChange, st.GenesisValidatorsRoot())
+	if err != nil {
+		return nil, err
+	}
+	for i, change := range changes {
+		if ctx.Err() != nil {
+			return nil, ctx.Err()
+		}
```

Comment:
Is checking context error needed? I think that's often done to while loop or recursive loops